frappe/contacts/doctype/contact/contact.py

```diff
@@ -122,6 +122,17 @@ def set_primary(self, fieldname):
 				setattr(self, fieldname, d.phone)
 				break
 
+	def before_save(self):
+		self.title = f'{self.first_name} {self.last_name or ""}'
+
+		if frappe.db.exists("Contact", self.title):
```

Comment:
- self.title is not the actual name!
```suggestion
		if frappe.db.exists("Contact", {"title": self.title}):
```

Reply:
fixed this change